### ISSUE TITLE
Headers are not handled well when combined with details/summary tags

### ISSUE THREAD

**Issue by alexei-osipov** [matthewwithanm/python-markdownify#129]
`h3` results in badly formatted output when combined with `details`/`summary` tags

This code
```
from bs4 import BeautifulSoup
from markdownify import MarkdownConverter

sample = """
<details><summary>Some question text</summary><div><h3>Answer</h3> Answer text</div></details>
"""

soup = BeautifulSoup(sample, 'html.parser')
result = MarkdownConverter().convert_soup(soup)
print(result)
```
Actual result (on `0.12.1`):
```
Some question text### Answer

 Answer text
```
Expected result:
```
Some question text

### Answer

Answer text
```

**Comment by jsm28** [comment]
PR #120 should help here, it fixes various cases of block tags not being properly separated from adjacent content, including headers.

**Comment by AlexVonB** [comment]
This was fixed with #120 , thank you for reporting!
